### ISSUE TITLE
Special Character

### ISSUE THREAD

**Issue by PointPubMedia** [jishi/node-sonos-http-api#544]
@jishi ,

Don't know why, but since yesterday, got an issue when trying french character.

Like

http://192.168.168.200:5005/Kitchen/say/Démarrage de mon interface/Julie/60

Got this:

{"status":"error","error":"Got status 400 when invoking /synthesize","stack":"Error\n    at promise.then (/opt/node-sonos-http-api/lib/tts-providers/microsoft.js:76:12)\n    at <anonymous>\n    at process._tickDomainCallback (internal/process/next_tick.js:228:7)"}

Not sure if that can help, but this morning I delete all my static file under TTS... looks like downloading from Microsoft with special character is crashing ...

Did another test, it's really coming from Microsoft.   I switch back to voicerss and it works..

**Comment by jishi** [comment]
Did this work before? 

Microsoft has made some backend changes, which for example triggered the 408 response if you didn't send content-length, and this might be some other requirements in how it should escape unicode characters, or how we should defined which character set we are sending.

**Comment by jishi** [comment]
When testing this in the Bing interface, it shows the generated SSML to be:
```
<speak version="1.0" xmlns="http://www.w3.org/2001/10/synthesis" xmlns:mstts="http://www.w3.org/2001/mstts" xml:lang="fr-FR"><voice xml:lang="fr-FR" name="Microsoft Server Speech Text to Speech Voice (fr-FR, Julie, Apollo)">D&#233;marrage de mon interface</voice></speak>
```
Which means that using entity encoding would resolve this issue, although the SSML standard dictates that you can use any UTF-8 characters (because UTF-8 should be default when no xml encoding attribute is present). Apparently they do not follow the spec 100%, but that should be easily resolvable.

**Comment by PointPubMedia** [comment]
@jishi :  I think I got this issue since a while.   After thinking about that, I have some "dynamic" message in the morning that are never the same, so I need to generate them and download them.   And most of the time I have some french character in it and the output sound never play totally and this morning, using voicerss I got my full dynamic message.

Looks like MS is having a hard time since a while ;)
